### PR TITLE
Creates a transaction mapper

### PR DESCRIPTION
**Pull request by BenjaminParment**
-> Adds transaction mapper
-> fetches transactions when mounting the transaction page
-> Change labelContent "In Progress" to "In progress" as this is how it is in the backend
-> Adds display when there are no transactions returned
-> Adds a formatter to display price of a transaction
-> Adds similar tests to yours to test the mapper and the formatters

### CODE REVIEW

File: src/components/transactions/Transactions.tsx [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
-type ConnectedTransactionsProps = ReturnType<typeof mapStateToProps>;
+const mapDispatchToProps = (dispatch: AppDispatch) => ({
+    fetchTransactions: () => dispatch(fetchTransactions()),
+});
+
+type ConnectedTransactionsProps = ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;
 
 class Transactions extends React.Component<ConnectedTransactionsProps> {
+    componentDidMount = () => {
+        this.props.fetchTransactions();
```
Comment (j-n-white) [inline src/components/transactions/Transactions.tsx]: I think we should be consistent with the submissions page as to whether we fetch every mount or only if we don't already have the data.

Also write a test fro this.

File: src/components/transactions/transactionsFormatters.test.ts [hunk 2]
```diff
@@ -29,3 +29,31 @@ describe("Given a call to format Transaction Number function", () => {
...
+    const amount = 50;
+    let currency: string;
+
+    describe("when called with GBP as a currency", () => {
+        beforeEach(() => {
+            currency = "GBP";
+        });
+
+        it(`it returns "£${amount}"`, () => {
+            expect(formatTransactionPrice(amount, currency)).toBe(`£${amount}`);
```
Comment (j-n-white) [inline src/components/transactions/transactionsFormatters.test.ts]: A minor thing. You have basically the same code here in the test as you do in the deployed code so all you are really testing here is that the same code does the same thing when copied to a different place.
```suggestion
            expect(formatTransactionPrice(amount, currency)).toBe("£50");

```
Also change the amount to have .00 to make the test more realistic.

File: src/components/transactions/transactionsFormatters.ts [hunk 3]
```diff
@@ -1,2 +1,10 @@
 export const formatTransactionsNumber = (dataSize?: number): string =>
     `${dataSize || 0} transaction${dataSize === 1 ? "" : "s"}`;
+
+export const formatTransactionPrice = (amount: string, currency: string): string => {
+    if (currency === "GBP") {
+        return `£${amount}`;
+    } else {
+        throw new Error("Currency is not GBP");
```
Comment (j-n-white) [inline src/components/transactions/transactionsFormatters.ts]:
```suggestion
        throw new Error(`${currency} is not supported for transactions. Only GBP is currently supported`);

```

File: src/components/transactions/Transactions.tsx (hunk 1 above)
Comment (BenjaminParment) [reply src/components/transactions/Transactions.tsx]: oh ur right, forgot about that oops

File: .env.development [hunk 4]
```diff
@@ -1,7 +1,7 @@
 PORT=30004
 REACT_APP_DYNAMIC_AUTH=false
 REACT_APP_COGNITO_REDIRECT_URL=http://localhost:30004
-REACT_APP_API_ENDPOINT=http://localhost/poc
+REACT_APP_API_ENDPOINT=http://localhost:3000
 REACT_APP_COGNITO_APP_CLIENT_ID=SampleClientId
-REACT_APP_COGNITO_ENDPOINT=http://localhost:30001
+REACT_APP_COGNITO_ENDPOINT=http://localhost:3000
```
Comment (j-n-white) [inline .env.development]: I'm assuming this isn't meant to be here

File: src/components/transactions/transactionsFormatters.test.ts [hunk 5]
```diff
@@ -31,16 +31,16 @@ describe("Given a call to format Transaction Number function", () => {
 });
 
 describe("Given a call to the format Transaction Price function", () => {
-    const amount = 50;
+    const amount = "50";
```
Comment (j-n-white) [inline src/components/transactions/transactionsFormatters.test.ts]: Include the pennies to make this more realistic data
```suggestion
    const amount = "50.00";

```

File: src/components/transactions/transactionsFormatters.test.ts [hunk 6]
```diff
@@ -31,16 +31,16 @@ describe("Given a call to format Transaction Number function", () => {
...
 
     describe("when called with GBP as a currency", () => {
         beforeEach(() => {
             currency = "GBP";
         });
 
-        it(`it returns "£${amount}"`, () => {
-            expect(formatTransactionPrice(amount, currency)).toBe(`£${amount}`);
+        it(`it returns "£50"`, () => {
+            expect(formatTransactionPrice(amount, currency)).toBe(`£50`);
```
Comment (j-n-white) [inline src/components/transactions/transactionsFormatters.test.ts]: does the linting like this I thought it preferred `"` then `'` if there are double quotes in the string?

Comment (BenjaminParment) [reply src/components/transactions/transactionsFormatters.test.ts]: lint doesnt seem to work so well on tests, prettier still formats it on save but thats pretty much it. it was the same in MTG

Comment (j-n-white) [reply src/components/transactions/transactionsFormatters.test.ts]: Well try and keep this consistent with how the deployed code is formatted

File: src/components/transactions/state/transactionsMapper.test.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import { DtoPage } from "shared/network/DtoPage";
+import { Page } from "state/page";
+import { TransactionRow } from "../transaction-table/TransactionTable";
+import moment from "moment";
+
+describe("Given a transactions mapper", () => {
+    const id1 = "1";
+    const id2 = "2";
+    const amount1 = "40";
+    const amount2 = "60";
```
Comment (BenjaminParment) [inline src/components/transactions/state/transactionsMapper.test.ts]: will change those to 40.00 and 60.00

File: src/components/transactions/transactionsFormatters.test.ts [hunk 8]
```diff
@@ -31,16 +31,16 @@ describe("Given a call to format Transaction Number function", () => {
...
 
     describe("when called with GBP as a currency", () => {
         beforeEach(() => {
             currency = "GBP";
         });
 
-        it(`it returns "£50"`, () => {
-            expect(formatTransactionPrice(amount, currency)).toBe(`£50`);
+        it('it returns "£50.00"', () => {
+            expect(formatTransactionPrice(amount, currency)).toBe(`£50.00`);
```
Comment (j-n-white) [inline src/components/transactions/transactionsFormatters.test.ts]: You still have the wrong quotes

Comment (BenjaminParment) [reply src/components/transactions/transactionsFormatters.test.ts]: wait, am i supposed to write "it returns '£50.00'" or 'it returns "£50.00"'

Comment (j-n-white) [reply src/components/transactions/transactionsFormatters.test.ts]: The latter, it's fine

**Pull request merged**